packages/utils/telemetry-utils/src/utils.ts

```diff
@@ -30,3 +32,74 @@ export function logIfFalse(
 	logger.send(newEvent);
 	return false;
 }
+
+/**
+ * Used in conjunction with the {@link createSampledLogger} to control logic for sampling events.
+ *
+ * @internal
+ */
+export interface IEventSampler<> {
+	/**
+	 * @returns true if the event should be sampled or false if not
+	 */
+	sample: () => boolean | undefined;
+}
+
+/**
+ * A telemetry logger that has sampling capabilities
+ *
+ * @internal
+ */
+export interface ISampledTelemetryLogger extends ITelemetryLoggerExt {
+	isSamplingDisabled: boolean;
+	eventSampler?: IEventSampler;
+}
+
+/**
+ * Wraps around an existing logger matching the {@link ITelemetryLoggerExt} interface and provides the ability to only log a subset of events using a sampling strategy.
+ * The sampling functionality uses the Fluid telemetry logging configuration along with the optionally provided event sampling callback to determine whether an event should
+ * be logged or not.
+ *
+ * Configuration object parameters:
+ * 'Fluid.Telemetry.DisableSampling': if this config value is set to true, all events will be unsampled and therefore logged.
+ * Otherwise only a sample will be logged according to the provided event sampler callback.
+ *
+ * @internal
+ */
+export function createSampledLogger(
+	logger: ITelemetryLoggerExt,
+	eventSampler?: IEventSampler,
+): ISampledTelemetryLogger {
+	const monitoringContext = loggerToMonitoringContext(logger);
+	const isSamplingDisabled = monitoringContext.config.getBoolean(
+		"Fluid.Telemetry.DisableSampling",
+	);
+
+	const sampledLogger = {
+		send: (event: ITelemetryBaseEvent): void => {
+			// if sampling is disabled, log all events. Otherwise, use the eventSampler to determine if the event should be logged.
+			if (isSamplingDisabled || (!isSamplingDisabled && eventSampler?.sample())) {
+				logger.send(event);
+			}
+		},
+		sendTelemetryEvent: (event: ITelemetryGenericEventExt): void => {
+			if (isSamplingDisabled || (!isSamplingDisabled && eventSampler?.sample())) {
+				logger.sendTelemetryEvent(event);
+			}
+		},
+		sendErrorEvent: (event: ITelemetryGenericEventExt): void => {
+			if (isSamplingDisabled || (!isSamplingDisabled && eventSampler?.sample())) {
+				logger.sendErrorEvent(event);
+			}
+		},
+		sendPerformanceEvent: (event: ITelemetryGenericEventExt): void => {
+			if (isSamplingDisabled || (!isSamplingDisabled && eventSampler?.sample())) {
+				logger.sendPerformanceEvent(event);
+			}
+		},
+		eventSampler,
+		isSamplingDisabled: isSamplingDisabled === true,
```

Comment:
right now the type for isSamplingDisabled when you get it from the config is actually `boolean | undefined`. I wasn't sure if it would actually be undefined or not so I just added the `=== true` there to ensure we have a boolean.

LMK if you still think we should change this

Reply:
Oh. If the typing says `getBoolean()` can return undefined, I actually vote for coalescing that call to `false`, so we're guaranteed to always have an actual boolean variable in this scope.

```typescript
const isSamplingDisabled = monitoringContext.config.getBoolean(
		"Fluid.Telemetry.DisableSampling",
	) ?? false;
```

Reply:
Updated in commit # c53a97